sdk/keyvault/keyvault-admin/test/public/accessControlClient.spec.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "chai";
+import { env, Recorder } from "@azure/test-utils-recorder";
+
+import { KeyVaultAccessControlClient } from "../../src";
+import { authenticate } from "../utils/authentication";
+
+describe("KeyVaultAccessControlClient", () => {
+  let client: KeyVaultAccessControlClient;
+  let recorder: Recorder;
+  let generateFakeUUID: () => string;
+  const globalScope = "/";
+
+  beforeEach(async function() {
+    const authentication = await authenticate(this);
+    client = authentication.accessControlClient;
+    recorder = authentication.recorder;
+    generateFakeUUID = authentication.generateFakeUUID;
+  });
+
+  afterEach(async function() {
+    await recorder.stop();
+  });
+
+  // The tests follow
+
+  it("listRoleDefinitions", async function() {
+    const expectedType = "Microsoft.Authorization/roleDefinitions";
+    let receivedRoles: string[] = [];
+
+    for await (const roleDefinition of client.listRoleDefinitions("/")) {
+      // Each role definition will have the shape of:
+      //
+      //   {
+      //     id: 'Microsoft.KeyVault/providers/Microsoft.Authorization/roleDefinitions/<ID>',
+      //     name: '<ID>',
+      //     type: '<role-type>',
+      //     roleName: '<role-name>',
+      //     // ...
+      //   }
+      //
+      assert.equal(roleDefinition.type, expectedType);
+      receivedRoles.push(roleDefinition.roleName!);
+    }
+
+    assert.deepEqual(receivedRoles, [
+      "Managed HSM Administrator",
+      "Managed HSM Crypto Officer",
+      "Managed HSM Crypto User",
+      "Managed HSM Policy Administrator",
+      "Managed HSM Crypto Auditor",
+      "Managed HSM Crypto Service Encryption",
+      "Managed HSM Backup"
+    ]);
+  });
+
+  it("createRoleAssignment and deleteRoleAssignment", async function() {
+    const roleDefinition = (await client.listRoleDefinitions(globalScope).next()).value;
+    const name = generateFakeUUID();
+    const assignment = await client.createRoleAssignment(
+      globalScope,
+      name,
+      roleDefinition.id!,
+      env.AZURE_TENANT_ID
```

Comment:
nit: should we make this a real objectID?

Reply:
@christothes this environment variable gets a real ObjectId through the `.env` file 🙂 the recordings end up replacing it with a fake value.

Reply:
That makes sense 😄 - I meant to say why `AZURE_TENANT_ID` and not `AZURE_OBJECT_ID`

Reply:
🤔 Tenant Ids do work though, at least form my tests. Perhaps I'm confusing things. Let's sync up on teams.

Reply:
The service will accept any uuid, it just won't work in allowing any principal to gain the created assignment. 😉

Reply:
Alright, I'm convinced 😅

Reply:
@danieljurek is this the tenant's object ID? Or is there another way of obtaining that one from the test pipelines?